src/js/view/components/slider.js

```diff
@@ -72,8 +72,10 @@ define([
                 }
             }
 
-            this.render(percentage);
-            this.update(percentage);
+            // the update method can set limits and return an updated value (TimeSlider)
+            var updatedPercent = this.update(percentage);
```

Comment:
Create a new method in Slider/TimeSlider for capping percentage value before passing it to render and update.